src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingSubmissionService.java

```diff
@@ -666,7 +665,7 @@ public List<ProgrammingSubmission> getAllProgrammingSubmissionsAssessedByTutorFo
             // Latest submission might be illegal
             submissions = participations.stream().map(StudentParticipation::findLatestLegalOrIllegalSubmission).filter(Optional::isPresent).map(Optional::get)
                     // filter out the submissions that don't have a result (but a null value) for the correctionRound
-                    .filter(submission -> submission.hasResultForCorrectionRound(correctionRound)).collect(toList());
+                    .filter(submission -> submission.hasResultForCorrectionRound(correctionRound)).collect(Collectors.toList());
```

Comment:
```suggestion
                    .filter(submission -> submission.hasResultForCorrectionRound(correctionRound)).toList();
```

Reply:
This is currently used in many more places and these two methods are in fact not the same, since .toList() returns an unmodifiable List while Collectors.toList() returns an ArrayList. Therefore I'd suggest changing this in a separate PR and verifying it there that the behavior stayed the same.

Reply:
I agree, but there are no guarantees at all :) https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/stream/Collectors.html#toList()

`There are no guarantees on the type, mutability, serializability, or thread-safety of the List returned`

Reply:
Exactly. But sadly some code snipets might currently rely on it non the less.
So I'd suggest moving this to an own PR where all occurrences are either replaced by .toList() or .toCollection(ArrayList::new)).

This PR is already large enough and I don't want to introduce more potential errors especially if they are not related to the PR